routes.js

```diff
@@ -43,27 +50,26 @@ function fastifySwagger (fastify, opts, next) {
     url: '/documentation',
     method: 'GET',
     schema: { hide: true },
-    handler: (request, reply) => reply.redirect(request.raw.url + '/')
+    handler: (request, reply) => reply.redirect('./documentation/')
   })
 
-  fastify.route({
-    url: '/documentation/:file',
-    method: 'GET',
-    schema: { hide: true },
-    handler: sendStaticFiles
+  // server swagger-ui with the help of fastify-static
+  fastify.register(require('fastify-static'), {
+    root: swaggerUiAssetPath,
+    prefix: `/documentation/`
   })
 
-  function sendStaticFiles (req, reply) {
-    if (!req.params.file) {
-      const file = files['index.html']
-      reply.type(file.type).send(file.contents)
-    } else if (files.hasOwnProperty(req.params.file)) {
-      const file = files[req.params.file]
-      reply.type(file.type).send(file.contents)
-    } else {
-      return reply.code(404).send(new Error('Not found'))
+  // hijak swagger index.html response
+  fastify.addHook('onSend', (request, reply, payload, next) => {
+    if (
+      request.raw.originalUrl === `/documentation/` ||
+      request.raw.originalUrl === `/documentation/index.html`
+    ) {
+      reply.header('Content-Type', 'text/html; charset=UTF-8')
+      payload = files.index
```

Comment:
this should not be needed, maybe fastify-static has a bug. Why do you need it?

Reply:
I don't think these prefixes are correct. I can mount this plugin with a given `/swagger`  and it will add the ui as `/swagger/documentation/`.

A better approach would be to either have a `post-install` script or a `prepublish`  script where we do these changes and replacements. A prepublish might be better, because we can't guarantee that regexps are stable across versions. Why do we need those changes anyway?

Reply:
This changes needed because by default it `index.html` of the swagger ui leads to the pet store.
https://github.com/swagger-api/swagger-ui/blob/master/dist/index.html

> I can mount this plugin with a given /swagger and it will add the ui as /swagger/documentation/.
how would you do that? I think `prefix` option is not supported.

I've tried this together with `example/dynamic.js`:
```js
fastify.register(require('../index'), {
  swagger: {
    info: {
      title: 'Test swagger',
      description: 'testing the fastify swagger api',
      version: '0.1.0'
    },
    host: 'localhost',
    schemes: ['http'],
    consumes: ['application/json'],
    produces: ['application/json']
  },
  exposeRoute: true,
  prefix: '/swagger'
})
```

the documentation appeared on `localhost:3000/documentation` still.

> a prepublish  script where we do these changes and replacements. 
you mean on prepublish we copy everything into our static folder and then serve it instead of the original `swagger-ui-dist`?

Reply:
Here is the new variant https://github.com/fastify/fastify-swagger/pull/39